packages/react-vapor/src/components/modal/examples/modalWindowExamples.tsx

```diff
@@ -118,13 +115,72 @@ const ModalExampleDisconnected: React.FunctionComponent<ModalConnectedExamplePro
 
 const ModalCompositeExampleConnected = connect(null, mapDispatchToProps)(ModalExampleDisconnected);
 
+const mapStateToProps = (state: IReactVaporState, ownProps: any) => ({
+    isDirty: WithDirtySelectors.getIsDirty(state, {id: ownProps.id}),
+});
+
+const ModalWithDirtyChangeDiscardPreventionDisconnected: React.FunctionComponent<ModalConnectedExampleProps> = ({
+    id,
+    close,
+    open,
+    toggleIsDirty,
+    isDirty,
+}) => {
+    const handleClose = () => {
+        close(id);
+        toggleIsDirty(id, false);
+    };
+
+    return (
+        <Section level={3} title="A modal implemented with the <UnsavedChangesModalProvider/>">
+            <Button className="btn" onClick={() => open(id)}>
+                Open Modal
+            </Button>
+            <UnsavedChangesModalProvider isDirty={isDirty}>
+                {({promptBefore}) => (
+                    <ModalCompositeConnected
+                        id={id}
+                        title="A modal with a dirty change discard prevention"
+                        classes={['mod-fade-in-scale']}
+                        modalBodyChildren={
+                            <div className="mt2">
+                                <div className="mb2">
+                                    <Input
+                                        id="input"
+                                        labelTitle="Try to close me with dirty changes."
+                                        onChange={(i) => toggleIsDirty(id, _.isEmpty(i) ? false : true)}
+                                    />
+                                </div>
+                                {loremIpsum({count: 10})}
+                            </div>
+                        }
+                        modalFooterChildren={
+                            <Button className="btn" onClick={() => promptBefore(() => handleClose()) && handleClose()}>
```

Comment:
why is handleClose called 2 times here?

Reply:
It is used as a callBack in the `promptBefore` method as the function to use if the user wants to exit the modal anyway even with dirty changes. But if the `promptBefore` returns true, it will use the method without further validation. It offers a lot of flexibility IMO.